UnitsNet/Serialization/Contracts/QuantityWithAbbreviationContract.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System.Runtime.Serialization;
+
+namespace UnitsNet.Serialization.Contracts
+{
+    /// <summary>
+    /// Can be used to represent a quantity, with it's unit defined using a common abbreviation (e.g. 'kg')
+    /// <remarks>
+    /// <para>Given the ambiguous nature of the unit-abbreviation mapping, an addition identifier, associated with the quantity type, is required</para>
+    /// </remarks>
+    /// </summary>
+    /// <typeparam name="TQuantity">The quantity type identifier (e.g. int / string) </typeparam>
+    /// <typeparam name="TValue">The type of value associated with this quantity (e.g. double / decimal) </typeparam>
+    [DataContract]
+    public class QuantityWithAbbreviationContract<TQuantity, TValue> : QuantityValueContract<TValue, string>
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="quantityType">>The quantity type identifier (e.g. int / string)</param>
+        /// <param name="value">The value</param>
+        /// <param name="unitAbbreviation">The associated unit abbreviation (e.g. 'kg') </param>
+        public QuantityWithAbbreviationContract(TQuantity quantityType, TValue value, string unitAbbreviation) : base(value, unitAbbreviation)
+        {
+            QuantityType = quantityType;
+        }
+
+        /// <summary>
+        /// The quantity type identifier.
+        /// </summary>
+        [DataMember(Order = 3)]
+        public TQuantity QuantityType { get; protected set; } 
```

Comment:
Codecov points out some of these data contract members may be lacking test coverage. If that is correct, I think we should add some basic serialization tests for these.